### PR TITLE
refactor: base properties configurable via registry

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

### What is the new behavior?

组件物料的基础属性可配置化

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Introduced `basePropertyOptions` to the materials plugin for enhanced property configurations.
  - Added a new function `patchBaseProps` to improve schema property management.

- **Bug Fixes**
  - Simplified Chinese language labels in various configuration files for better clarity.

- **Documentation**
  - Updated Chinese labels in JSON files to reflect concise terminology.

### CODE REVIEW

**Comment by gene9831** [comment]
@coderabbitai review this pull request

**Comment by gene9831** [comment]
@coderabbitai review

File: packages/plugins/materials/src/js/options.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+export const baseProperties = [
```
Comment (hexqi) [inline packages/plugins/materials/src/js/options.js]: 要考虑下分组，还有添加的位置是在分组的开始还是结束，这个格式要调整下

Comment (gene9831) [reply packages/plugins/materials/src/js/options.js]: updated

**Comment by gene9831** [comment]
@coderabbitai review

**Review by hexqi — DISMISSED** [review]

File: packages/plugins/materials/src/composable/useMaterial.js [hunk 2]
```diff
@@ -76,11 +77,41 @@ const getConfigureMap = () => {
...
+const patchBaseProps = (schemaProperties) => {
+  if (!Array.isArray(schemaProperties)) {
+    return
+  }
+
+  const { properties = [], insertPosition = 'end' } = getOptions(meta.id).basePropertyOptions || {}
+
+  for (const props of properties) {
+    // TODO 判断分组相同，是否有其他更好的方法
+    const group = schemaProperties.find((item) => item.label.zh_CN === props.label.zh_CN)
```
Comment (hexqi) [inline packages/plugins/materials/src/composable/useMaterial.js]: 可以添加一个group字段，查找group一致或者label.zh_CN一致

Comment (rhlin) [reply packages/plugins/materials/src/composable/useMaterial.js]: 对比zh_CN感觉有点怪， 有没有其他key值可以对比的

Comment (gene9831) [reply packages/plugins/materials/src/composable/useMaterial.js]: > 对比zh_CN感觉有点怪， 有没有其他key值可以对比的

目前schema协议中并没有其他key可以对比。方案是扩展新增一个group字段，并且也对比label.zh_CN做兼容处理

**Review by hexqi — DISMISSED** [review]

**Comment by hexqi** [comment]
image
当前bundle物料文件里面，部分组件已经存在“其他”、“其他配置”、“其他属性”的分组，会出现多个其他分组，可以把这块都统一为“其他”

**Review by hexqi — APPROVED** [review]

**Pull request merged**